src/postgres/src/test/regress/expected/yb_pg_foreign_data.out

```diff
@@ -0,0 +1,2268 @@
+ERROR:  column "c2" in child table must be marked NOT NULL
+ALTER FOREIGN TABLE fd_pt2_1 ALTER c2 SET NOT NULL;
+ALTER TABLE fd_pt2 ATTACH PARTITION fd_pt2_1 FOR VALUES IN (1);
+ALTER TABLE fd_pt2 DETACH PARTITION fd_pt2_1;
+ALTER TABLE fd_pt2 ADD CONSTRAINT fd_pt2chk1 CHECK (c1 > 0);
+\d+ fd_pt2
+                                   Table "public.fd_pt2"
+ Column |  Type   | Collation | Nullable | Default | Storage  | Stats target | Description 
+--------+---------+-----------+----------+---------+----------+--------------+-------------
+ c1     | integer |           | not null |         | plain    |              | 
+ c2     | text    |           | not null |         | extended |              | 
+ c3     | date    |           |          |         | plain    |              | 
+Partition key: LIST (c1)
+Check constraints:
+    "fd_pt2chk1" CHECK (c1 > 0)
+Number of partitions: 0
+
+\d+ fd_pt2_1
+                                     Foreign table "public.fd_pt2_1"
+ Column |  Type   | Collation | Nullable | Default | FDW options | Storage  | Stats target | Description 
+--------+---------+-----------+----------+---------+-------------+----------+--------------+-------------
+ c1     | integer |           | not null |         |             | plain    |              | 
+ c2     | text    |           | not null |         |             | extended |              | 
+ c3     | date    |           | not null |         |             | plain    |              | 
+Check constraints:
+    "p21chk" CHECK (c2 <> ''::text)
+Server: s0
+FDW options: (delimiter ',', quote '"', "be quoted" 'value')
+
+ALTER TABLE fd_pt2 ATTACH PARTITION fd_pt2_1 FOR VALUES IN (1);       -- ERROR
+ERROR:  child table is missing constraint "fd_pt2chk1"
+ALTER FOREIGN TABLE fd_pt2_1 ADD CONSTRAINT fd_pt2chk1 CHECK (c1 > 0);
+ALTER TABLE fd_pt2 ATTACH PARTITION fd_pt2_1 FOR VALUES IN (1);
+-- TRUNCATE doesn't work on foreign tables, either directly or recursively
+TRUNCATE fd_pt2_1;  -- ERROR
+ERROR:  "fd_pt2_1" is not a table
+TRUNCATE fd_pt2;  -- ERROR
+ERROR:  "fd_pt2_1" is not a table
+DROP FOREIGN TABLE fd_pt2_1;
+DROP TABLE fd_pt2;
+-- foreign table cannot be part of partition tree made of temporary
+-- relations.
+CREATE TEMP TABLE temp_parted (a int) PARTITION BY LIST (a);
+CREATE FOREIGN TABLE foreign_part PARTITION OF temp_parted DEFAULT
+  SERVER s0;  -- ERROR
+ERROR:  cannot create a permanent relation as partition of temporary relation "temp_parted"
+CREATE FOREIGN TABLE foreign_part (a int) SERVER s0;
+ALTER TABLE temp_parted ATTACH PARTITION foreign_part DEFAULT;  -- ERROR
+ERROR:  cannot attach a permanent relation as partition of temporary relation "temp_parted"
+DROP FOREIGN TABLE foreign_part;
+DROP TABLE temp_parted;
+-- Cleanup
+DROP SCHEMA foreign_schema CASCADE;
+DROP ROLE regress_test_role;                                -- ERROR
+ERROR:  role "regress_test_role" cannot be dropped because some objects depend on it
+DETAIL:  privileges for foreign-data wrapper foo
```

Comment:
I have no idea why but this has a tendency to arrive in a different order.

This + the `DETAIL` at line 444 and 445 makes this test fail almost every time.

Reply:
I think the reason for this is hidden in the implementation of `ybc_systable_beginscan`. The reason why I think this is the case is:

- the error message `cannot be dropped because some objects depend on it` exists in [`user.c`](https://github.com/yugabyte/yugabyte-db/blob/master/src/postgres/src/backend/commands/user.c#L1067)
- which calls [`checkSharedDependencies`](https://github.com/yugabyte/yugabyte-db/blob/master/src/postgres/src/backend/catalog/pg_shdepend.c#L553)
- that function works on the result of [`systable_beginscan`](https://github.com/yugabyte/yugabyte-db/blob/master/src/postgres/src/backend/access/index/genam.c#L334) which in case of YugabyteDB is redefined to:

```c
	if (IsYugaByteEnabled())
	{
		return ybc_systable_beginscan(heapRelation,
		                              indexId,
		                              indexOK,
		                              snapshot,
		                              nkeys,
		                              key);
	}
```

My assumption is: Postgres works on an ordered _scan_ while YugabyteDB implementation appears to be working on a random order _scan_.

Reply:
To me, the most reasonable approach here would be to string the detail string by `\n` and sort the resulting array, then join back. It may not be fully compatible with Postgres but at least the order will be predictive. Unless this problem surfaces some underlying index storage problem...